packages/nx/src/adapter/ngcli-adapter.ts

```diff
@@ -503,7 +503,10 @@ export async function generate(
   verbose: boolean
 ) {
   const logger = getLogger(verbose);
-  const fsHost = new NxScopedHost(root);
+  const fsHost = new NxScopeHostUsedForWrappedSchematics(
+    root,
+    new FsTree(root, verbose)
+  );
```

Comment:
Is the original NxScopedHost still used anywhere? Maybe we can remove it and only keep the sync implementation

Reply:
Yeah, it's used for executors. The `read` operation is faster using the project graph and executors don't write back. So there's no issue with the async write implementation. That said, I plan to refactor/rename our existing host implementations, so they better represent what they are for. I'll do it in a follow-up PR later.